File: AlbumLibrary/src/main/java/com/album/presenter/AlbumPresenter.java
package com.album.presenter;

import com.album.entity.AlbumEntity;

import java.util.ArrayList;

/**
 * by y on 14/08/2017.
 */

public interface AlbumPresenter {
    void scan(String bucketId, int page, int count);

    void mergeEntity(ArrayList<AlbumEntity> albumList, ArrayList<AlbumEntity> multiplePreviewList);

    void firstMergeEntity(ArrayList<AlbumEntity> albumEntityList, ArrayList<AlbumEntity> selectEntity);

    boolean isScan();

    void resultScan(String path);
}

File: AlbumLibrary/src/main/java/com/album/presenter/AlbumPresenter.kt
package com.album.presenter

import com.album.entity.AlbumEntity
import java.util.*

/**
 * by y on 14/08/2017.
 */

interface AlbumPresenter {

    fun getScanLoading(): Boolean

    fun scan(bucketId: String, page: Int, count: Int)

    fun mergeSelectEntity(albumList: ArrayList<AlbumEntity>, multiplePreviewList: ArrayList<AlbumEntity>)

    fun firstMergeEntity(albumEntityList: ArrayList<AlbumEntity>, selectEntity: ArrayList<AlbumEntity>?)

    fun resultScan(path: String)
}
